reports/generate_report_emails.py

```diff
@@ -55,58 +54,45 @@
     )
 )
 report_emails
+print(report_emails["report_url"])
 
 
-# +
-# generate mjml template
-stream = os.popen("npx mjml ../templates/email/report.mjml -s")
-mjml_template = stream.read()
-
-
-# for each row in report_emails, populate template
-def _generate_template(report_url):
-    # render variables in template which use e.g. {{ month_name }} by passing in config
-    completed = jinja_env.from_string(mjml_template).render(**config, url=report_url)
-    return completed
-
-
-html_messages = report_emails.report_url.apply(_generate_template)
-all_emails = report_emails.main_email
-all_emails_list = list(zip(all_emails, html_messages))
-
-print("Using server token (Sandbox is f38...): " + config["postmark_server_token"])
+print("Using server token: " + config["hubspot_server_token"])
 # +
 # Double check if we're on development that it is going to the sandbox ----
 if config.getboolean("is_development"):
-    server = postmark.server.get()
-    assert server.DeliveryType == "Sandbox"
-    print("In sandbox environment")
+    print("Using development values from config.ini")
 else:
-    print("In production environment")
+    print("Using production values from config.ini")
 
 # Prompt user on whether to continue (if show_prompt specified) ----
 if config.getboolean("show_prompt"):
     result = input(
         f"""
-You are about to email the following {all_emails.count()} addresses: {", ".join(all_emails)}
+You are about to email the following {report_emails["main_email"].count()} addresses: {", ".join(report_emails.main_email)}
 To continue, type yes."""
     )
     if result != "yes":
         raise Exception("Need yes to continue")
 
-
-for emails, html_messages in all_emails_list:
-    email = postmark.emails.Email(
-        From=config["email_from"],
-        To=emails,
-        Cc=config["cc_email"],
-        Subject=config["email_subject"],
-        HtmlBody=html_messages,
-        TrackOpens=config["track_opens"],
-        TrackLinks=config["track_links"],
+for index, email in report_emails.iterrows():
```

Comment:
just curious how long does this take to run? is there no multi-send with parameters?

Reply:
@atvaccaro I have only tested with two emails.... but I timed it and it took less than two seconds if that's helpful? I think if we want to do multi-send we need to [set up the SMTP server as per these docs](https://developers.hubspot.com/docs/api/marketing/transactional-emails#smtp-api)?